src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    if (this.hidden) {
+      this.hidden = false;
+    } else {
+      this.hidden = true;
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    const victimIndex = Animal.alive.findIndex(animal => animal === victim);
+
+    if (victim.hasOwnProperty('hidden') && victim.hidden === false) {
+      victim.health -= 50;
```

Comment:
You decrease the victim's health, but there is no check if the victim's health is already 0 or less. It would be better to add a condition to check if the victim's health is more than 0 before decreasing it.